kairon/shared/actions/utils.py

```diff
@@ -371,6 +371,28 @@ def get_slot_type(bot: str, slot: str):
             logger.exception(e)
             raise ActionFailure(f'Slot not found in database: {slot}')
 
+    @staticmethod
+    def perform_google_search_globally(search_term, **kwargs):
+        from googlesearch import search
+
+        results = []
+        try:
+            search_results = search(search_term, advanced=True, **kwargs)
```

Comment:
search() returns a generator object in which each element is of SearchResult type object and cannot be accessed like a dict

Reply:
eg: 
```
search_results = search("what is kanban?", advanced=True)
s = next(search_results)
```
SearchResult(url=https://kanbanize.com/kanban-resources/getting-started/what-is-kanban, title=What Is Kanban? Explained in 10 Minutes, description=Kanban is a popular Lean workflow management method for defining, managing, and improving services that deliver knowledge work. It helps you visualize work, ...)

Reply:
also take the website on which search has to be enabled in config and append this website in the search term as discussed earlier.
eg:
search_term = "How to add task plan"
website = "https://nimblework.com"
site_specific_search_term = f"{search_term}  site: {website}"